### PR TITLE
tutorial: Describe weight overflows and how to avoid them.

### PR DESCRIPTION
**Pull request by blp**
Thanks to Leonid Ryzhyk and Mihai Budiu for leading me to the root
cause.

Thanks to Ihar Hrachyshka <ihrachys@redhat.com> for reporting the
bug in OVN that prompted this.  Ihar's original report is at:
https://mail.openvswitch.org/pipermail/ovs-dev/2021-May/382967.html

### CODE REVIEW

**Comment by blp** [comment]
Leonid pointed out that my statement that you can insert duplicate records in relations was wrong. I force-pushed a fix that drops that statement.

File: doc/tutorial/tutorial.md [hunk 1]
```diff
@@ -2975,6 +2975,158 @@ These 5 kinds of statements (`for`, `if`, `match`, `var`, blocks enclosed in bra
...
+If we joined `Nationality` and `Occupation` in a rule, then the output
+record's weight would be the product of the input records' weights.
+In a pathological case where all `N` Persons had a single nationality
+and occupation, the output record would have weight `N^2`.
+
+Consider the following additional example, which is abstracted from an
+actual bug in a system built with DDlog.  Suppose that a networking
+system has a collection of switches, each of which may have ports with
+different types and names, as well as access control lists (ACLs) and
+load balacners.  The management system might provide the configuration
```
Comment (numansiddique) [inline doc/tutorial/tutorial.md:3025]: s/balacners/balancers

**Review by numansiddique — COMMENTED** [review]
Found  a small typo while reading the tutorial

**Comment by ryzhyk** [comment]
Let me see if I can resolve #878.  We can then update these docs to say that the program will crash with numeric overflow rather than just to expect incorrect behavior...

**Comment by blp** [comment]
@numansiddique Thanks for the spelling error fix. It didn't get folded in so I submitted pull #979 to get this and a few other typos fixed.

**Pull request merged**